python-checks/src/test/java/org/sonar/python/checks/SkLearnEstimatorDontInitializeEstimatedValuesCheckTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks;
+
+import org.junit.jupiter.api.Test;
+import org.sonar.python.checks.quickfix.PythonQuickFixVerifier;
+import org.sonar.python.checks.utils.PythonCheckVerifier;
+
+class SkLearnEstimatorDontInitializeEstimatedValuesCheckTest {
+  @Test
+  void test() {
+    PythonCheckVerifier.verify("src/test/resources/checks/sklearn_estimator_underscore_initialization.py", new SkLearnEstimatorDontInitializeEstimatedValuesCheck());
+  }
+
+  @Test
+  void testQuickfix1() {
+    PythonQuickFixVerifier.verify(
+      new SkLearnEstimatorDontInitializeEstimatedValuesCheck(),
+      """
+        from sklearn.base import BaseEstimator
+        class InheritingEstimator(BaseEstimator):
+            def __init__(self) -> None:
+                self.a_ = None
+                ...""",
+      """
+        from sklearn.base import BaseEstimator
+        class InheritingEstimator(BaseEstimator):
+            def __init__(self) -> None:
+                ..."""
+          );
+  }
+  @Test
+  void testQuickfixEmptyFunc() {
+    PythonQuickFixVerifier.verify(
+      new SkLearnEstimatorDontInitializeEstimatedValuesCheck(),
+      """
+        from sklearn.base import ClassifierMixin
+        class InheritingEstimator(ClassifierMixin):
+            def __init__(self) -> None:
+                self.a_ = None""",
+      """
+        from sklearn.base import ClassifierMixin
+        class InheritingEstimator(ClassifierMixin):
+            def __init__(self) -> None:
+                pass"""
```

Comment:
Maybe we could have a variation of the quickfix. One version would be remove and the other could be rename. The rename would remove the "_" I guess.